push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/serviceLease/ApnsServiceHolder.java

```diff
@@ -35,26 +43,43 @@
  *
  * This cache also listens for {@link VariantCompletedEvent} event and stops all instantiated {@link ApnsService}s and frees the cache.
  *
- * @see AbstractServiceCache#dequeueOrCreateNewService(String, String, ServiceConstructor)
- * @see AbstractServiceCache#queueFreedUpService(String, String, Object)
- * @see AbstractServiceCache#freeUpSlot(String, String)
+ * @see AbstractServiceHolder#dequeueOrCreateNewService(String, org.jboss.aerogear.unifiedpush.api.Variant, org.jboss.aerogear.unifiedpush.message.cache.AbstractServiceHolder.ServiceConstructor)
+ * @see AbstractServiceHolder#queueFreedUpService(String, org.jboss.aerogear.unifiedpush.api.Variant, Object)
+ * @see AbstractServiceHolder#freeUpSlot(String, org.jboss.aerogear.unifiedpush.api.Variant)
  */
 @ApplicationScoped
-public class ApnsServiceCache extends AbstractServiceCache<ApnsService> {
+public class ApnsServiceHolder extends AbstractServiceHolder<ApnsService> {
 
-    private final AeroGearLogger logger = AeroGearLogger.getInstance(ApnsServiceCache.class);
+    private final AeroGearLogger logger = AeroGearLogger.getInstance(ApnsServiceHolder.class);
 
     public static final int INSTANCE_LIMIT = 10;
-    public static final long INSTANCE_ACQUIRING_TIMEOUT = 5000;
+    public static final long INSTANCE_ACQUIRING_TIMEOUT = 7500;
+    public static final long DISPOSING_DELAY = 5000;
 
     @Inject
     private ClientInstallationService clientInstallationService;
 
-    public ApnsServiceCache() {
-        super(INSTANCE_LIMIT, INSTANCE_ACQUIRING_TIMEOUT);
+    @Resource(mappedName = "java:/queue/APNsBadgeLeaseQueue")
+    private Queue apnsBadgeLeaseQueue;
```

Comment:
@matzew in https://github.com/lfryc/aerogear-unifiedpush-server/pull/16#discussion_r56665027
> badge... is that a good name ? :)